lib/app/lifecycle.coffee

```diff
@@ -60,6 +61,7 @@ crypto = require 'crypto'
 @beforeSocialAuth = (provider) -> (req, res, next) ->
   req.session.redirectTo = req.query['redirect-to']
   req.session.skipOnboarding = req.query['skip-onboarding']
+  req.query['signup-intent'] && req.session.signupIntent = req.query['signup-intent']
```

Comment:
this line reads weird to me, did you mean just `req.session.signupIntent = req.query['signup-intent']` ?

Reply:
ah yeah, i'm just not used to idiomatic coffeescript so i've been very inconsistent with how i do these conditional assignments. on that note i'm not sure if i'm doing them correctly at all. my goal was to pass nothing if it is not there.

Reply:
then `req.session.signupIntent = req.query['signup-intent'] if req.query['signup-intent']` kinda thing.

Since you are checking `if req.session.signupIntent` later on you may just be fine to assign it here though.